com.unity.render-pipelines.universal/Runtime/Passes/MainLightShadowCasterPass.cs

```diff
@@ -121,9 +119,15 @@ public bool Setup(ref RenderingData renderingData)
 
         public override void Configure(CommandBuffer cmd, RenderTextureDescriptor cameraTextureDescriptor)
         {
-            m_MainLightShadowmapTexture = ShadowUtils.GetTemporaryShadowTexture(m_ShadowmapWidth,
-                m_ShadowmapHeight, k_ShadowmapBufferBits);
-            ConfigureTarget(new RenderTargetIdentifier(m_MainLightShadowmapTexture), GraphicsFormat.ShadowAuto, m_ShadowmapWidth, m_ShadowmapHeight, 1, true);
+            if (m_MainLightShadowmap == null || m_MainLightShadowmap.rt.width != m_ShadowmapWidth ||
+                m_MainLightShadowmap.rt.height != m_ShadowmapHeight)
+            {
+                m_MainLightShadowmap?.Release();
+                m_MainLightShadowmap = ShadowUtils.AllocShadowRTHandle(m_ShadowmapWidth, m_ShadowmapHeight,
+                    k_ShadowmapBufferBits, "_MainLightShadowmapTexture");
+            }
```

Comment:
Textures should be created and disposes per-camera, otherwise you keep resources around for no reason.